include/highfive/bits/H5Node_traits_misc.hpp

```diff
@@ -162,6 +162,78 @@ inline std::string NodeTraits<Derivate>::getObjectName(size_t index) const {
+    const size_t maxLength = 255;
+    char  buffer[maxLength + 1];
+    ssize_t retcode = H5Iget_name(
+        static_cast<const Derivate*>(this)->getId(), buffer, static_cast<hsize_t>(maxLength) + 1 );
+    if (retcode < 0) {
+        HDF5ErrMapper::ToException<GroupException>("Error accessing object name");
+    }
+    const size_t length = static_cast<std::size_t>(retcode);
+        if (length <= maxLength) {
+        return std::string(buffer, length);
+    }
+    std::vector<char> bigBuffer(length + 1, 0);
+    H5Iget_name(
+        static_cast<const Derivate*>(this)->getId(), bigBuffer.data(), static_cast<hsize_t>(length) + 1 );
+    return std::string(bigBuffer.data(), length);
+}
+
+template <typename Derivate>
+inline bool NodeTraits<Derivate>::moveObject(const std::string& src_name,
+                                             const std::string& dst_path, bool parents) const {
+    RawPropertyList<PropertyType::LINK_CREATE> lcpl;
+    if (parents) {
+        lcpl.add(H5Pset_create_intermediate_group, 1u);
+    }
+    herr_t status = H5Lmove(static_cast<const Derivate*>(this)->getId(), src_name.c_str(),
+                            static_cast<const Derivate*>(this)->getId(), dst_path.c_str(), lcpl.getId(), H5P_DEFAULT);
+    if (status  < 0) {
+        HDF5ErrMapper::ToException<GroupException>(
+                    std::string("Unable to move link to \"") + dst_path + "\":");
+        return false;
+    }
+    return true;
+}
+
+template <typename Derivate>
+inline bool NodeTraits<Derivate>::moveObject(const File& dst_parent, const std::string& src_name,
+                                             const std::string& dst_path, bool parents) const {
+    RawPropertyList<PropertyType::LINK_CREATE> lcpl;
+    if (parents) {
+        lcpl.add(H5Pset_create_intermediate_group, 1u);
+    }
+    herr_t status = H5Lmove(static_cast<const Derivate*>(this)->getId(), src_name.c_str(), dst_parent.getId(), dst_path.c_str(), lcpl.getId(), H5P_DEFAULT);
+    if (status  < 0) {
+        HDF5ErrMapper::ToException<GroupException>(
+                    std::string("Unable to move link to \"") + dst_path + "\":");
+        return false;
+    }
+    return true;
+}
+
+template <typename Derivate>
+inline bool NodeTraits<Derivate>::moveObject(const Group& dst_parent, const std::string& src_name,
+                                             const std::string& dst_path, bool parents) const {
+    RawPropertyList<PropertyType::LINK_CREATE> lcpl;
+    if (parents) {
+        lcpl.add(H5Pset_create_intermediate_group, 1u);
+    }
+    herr_t status = H5Lmove(static_cast<const Derivate*>(this)->getId(), src_name.c_str(), dst_parent.getId(), dst_path.c_str(), lcpl.getId(), H5P_DEFAULT);
+    if (status  < 0) {
+        HDF5ErrMapper::ToException<GroupException>(
+                    std::string("Unable to move link to \"") + dst_path + "\":");
+        return false;
+    }
+    return true;
+}
+
+
+
+
```

Comment:
```suggestion
template <typename Derivate>
inline bool NodeTraits<Derivate>::moveObject(const Group& group,
                                             const std::string& src_path,
                                             const std::string& dst_path,
                                             bool parents) const {
    RawPropertyList<PropertyType::LINK_CREATE> lcpl;
    if (parents) {
        lcpl.add(H5Pset_create_intermediate_group, 1u);
    }
    herr_t status = H5Lmove(group.getId(), src_path.c_str(),
                            group.getId(), dst_path.c_str(),
                            lcpl.getId(), H5P_DEFAULT);
    if (status < 0) {
        HDF5ErrMapper::ToException<GroupException>(
            std::string("Unable to move link to \"") + dst_path + "\":");
        return false;
    }
    return true;
}
```
See suggestion above